src/coreclr/vm/interpreter.cpp

```diff
@@ -9264,6 +9264,23 @@ void Interpreter::DoCallWork(bool virtualCall, void* thisArg, CORINFO_RESOLVED_T
             m_curStackHt++; didIntrinsic = true;
             break;
 #endif // INTERP_ILSTUBS
+
+        case NI_System_Runtime_CompilerServices_RuntimeHelpers_IsReferenceOrContainsReferences:
+        {
+            CORINFO_SIG_INFO sigInfoFull;
+            {
+                GCX_PREEMP();
+                m_interpCeeInfo.getMethodSig(CORINFO_METHOD_HANDLE(methToCall), &sigInfoFull, nullptr);
+            }
+
+            MethodTable* typeArg = GetMethodTableFromClsHnd(sigInfoFull.sigInst.methInst[0]);
+            OpStackSet<BOOL>(m_curStackHt, typeArg->ContainsPointers());
```

Comment:
Nit: This won't return the correct value for byref-like types

Reply:
Ah, you mean if IsReferenceOrContainsReference suddenly starts to accept byref-like types in its generic T? Fixed via https://github.com/dotnet/runtime/pull/103326/commits/83f5514cf9a0565c830788a5d857a5b75ada11fd

Reply:
IsReferenceOrContainsReference  accepts byref-like types as of last week: https://github.com/dotnet/runtime/pull/103010 . 

The implementation needs to recursively check whether there are byref fields in the byref-like type. It is not correct to just return true for any byref-like type. (It is ok to just add TODO here if you do not want to spend time implementing it correctly.)

Reply:
Addressed in https://github.com/dotnet/runtime/pull/103326/commits/f35e849f8195e1a87e0a9108ea645f56cf572270